src/vi/mimihentai/src/eu/kanade/tachiyomi/extension/vi/mimihentai/MiMiHentai.kt

```diff
@@ -139,36 +142,80 @@ class MiMiHentai : HttpSource() {
 
     override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
         val url = apiUrl.toHttpUrl().newBuilder().apply {
-            addPathSegments("search")
-            addQueryParameter("page", page.toString())
+            addPathSegments("advance-search")
             addQueryParameter("name", query)
-
+            addQueryParameter("page", maxOf(0, page - 1).toString())
             (if (filters.isEmpty()) getFilterList() else filters).forEach { filters ->
                 when (filters) {
                     is SortByList ->
                         {
                             val sort = getSortByList()[filters.state]
                             addQueryParameter("sort", sort.id)
                         }
-
+                    is Author -> addQueryParameter("author", filters.state)
+                    is Character -> addQueryParameter("character", filters.state)
+                    is Parody -> addQueryParameter("parody", filters.state)
+                    is GenreList -> addQueryParameter("genre", filters.toGenre().toString())
                     else -> {}
                 }
             }
         }.build()
         return GET(url, headers)
     }
 
+    private fun genresRequest(): Request = GET("$apiUrl/genres", headers)
+
+    private fun parseGenres(response: Response): List<Pair<Long, String>> {
+        return response.parseAs<List<Genres>>().map { Pair(it.id, it.name) }
+    }
+
+    private var fetchGenresAttempts: Int = 0
+    private fun fetchGenres() {
+        if (fetchGenresAttempts < 3 && genreList.isEmpty()) {
+            try {
+                val response = client.newCall(genresRequest()).execute()
```

Comment:
Inside a suspend function, use `await()` from extensions-lib instead of `execute()`.

```suggestion
                val response = client.newCall(genresRequest()).await()
```

Reply:
This is not resolved.

Reply:
> This is not resolved.

ok i used `await()`